server/text_generation_server/utils/logits_process.py

```diff
@@ -114,6 +114,8 @@ def __call__(self, input_ids: torch.Tensor, scores: torch.Tensor) -> torch.Tenso
             score < 0, score * self.penalty_tensor, score / self.penalty_tensor
         )
 
+        # set score to 0 where input_ids is a padding token
+        score *= input_ids.ne(0)
```

Comment:
```suggestion
```

I don't think this is needed for this one as we do not do a scatter_add but just a scatter (we only take a single value).

Reply:
that makes sense, great catch!